### PR TITLE
UI for CSV Upload

### PR DESCRIPTION
**Pull request by klanmiko**

### CODE REVIEW

**Comment by klanmiko** [comment]
fixes #23

File: applications-frontend/src/components/Upload.vue [hunk 1]
```diff
@@ -0,0 +1,84 @@
+<template>
+  <div class="container">
+    <progress class="loading is-primary" v-if="waiting" max="100"></progress>
+    <form v-else class="form" enctype="multipart/form-data" method="POST">
+      <div class="field">
+        <label class="label">Applicants: </label>
+        <div class="file has-name">
+          <label class="file-label">
+            <input id="file" class="file-input" type="file" @change="fileChanged" required name="resume" />
```
Comment (thomasmunduchira) [inline applications-frontend/src/components/Upload.vue]: `resume` shouldn't be the name of the element. Maybe `applicants_spreadsheet`?

File: applications-frontend/src/router.js [hunk 2]
```diff
@@ -3,10 +3,12 @@ import Router from "vue-router";
 import Applicant from "./components/Applicant";
 import Home from "./components/Home";
 import Configure from './components/Configure';
+import Upload from './components/Upload';
 import Settings from './components/Settings';
 import Error from "./Error";
 import VueResource from "vue-resource";
 import VueCookie from "vue-cookie";
+import { faUpload } from "@fortawesome/free-solid-svg-icons";
```
Comment (thomasmunduchira) [inline applications-frontend/src/router.js]: `faUpload` is not used in this file, remove.

File: applications-frontend/src/router.js [hunk 3]
```diff
@@ -47,7 +49,17 @@ let router = new Router({
     },
     {
       path: "/configure",
-      component: Configure
+      component: Configure,
+      meta: {
```
Comment (thomasmunduchira) [inline applications-frontend/src/router.js]: Can you add `requiresAuth: true` to the Settings route as well while you're at it?

File: src/config/config.py [hunk 4]
```diff
@@ -20,3 +20,4 @@ class Config:
     SQLALCHEMY_DATABASE_URI = 'postgresql://{}:{}@{}:{}/{}'.format(DB_USER, DB_PASSWORD,
                                                                    DB_HOSTNAME, DB_PORT, DB_NAME)
     USE_SESSION_FOR_NEXT = 1
+    UPLOAD_FOLDER = os.path.dirname(os.path.join(os.path.curdir, "../applicants"))
```
Comment (thomasmunduchira) [inline src/config/config.py]: `os.path.dirname(os.path.join(os.getcwd(), 'applicants'))`?

Comment (thomasmunduchira) [reply src/config/config.py]: A bit cleaner, if this is what you're shooting for

File: src/controller/admin.py [hunk 5]
```diff
@@ -20,14 +22,28 @@
...
+@admin.route('/api/admin/load', methods=['POST'])
 @login_required
 def load():
     """Reload applications from CSV file."""
     if current_user.role != Role.admin:
         abort(401, 'User needs to be an admin to access this route')
 
-    with open(os.path.join(os.getcwd(), 'sample.csv'), encoding='utf-8') as csv_file:
+    if 'applicants' not in request.files:
+        abort(401, 'No file submitted')
```
Comment (thomasmunduchira) [inline src/controller/admin.py]: Should be 400

File: src/controller/admin.py [hunk 6]
```diff
@@ -20,14 +22,28 @@
...
+    file = request.files['applicants']
+    # if user does not select file, browser also
+    # submit an empty part without filename
+    if file.filename == '':
+        abort(401, 'Invalid file name')
```
Comment (thomasmunduchira) [inline src/controller/admin.py]: Should be 400

File: src/controller/admin.py [hunk 7]
```diff
@@ -55,15 +71,32 @@ def load():
...
+@admin.route('/api/admin/reload', methods=['POST'])
 @login_required
 def reload():
     """Reload applications from CSV file."""
     if current_user.role != Role.admin:
         abort(401, 'User needs to be an admin to access this route')
 
-    with open(os.path.join(os.getcwd(), 'sample.csv'), encoding='utf-8') as csv_file:
+    if 'applicants' not in request.files:
+        abort(401, 'No file submitted')
```
Comment (thomasmunduchira) [inline src/controller/admin.py]: Should be 400

File: src/controller/admin.py [hunk 8]
```diff
@@ -55,15 +71,32 @@ def load():
...
+
+    file = request.files['applicants']
+    # if user does not select file, browser also
+    # submit an empty part without filename
+    if file.filename == '':
+        abort(401, 'Invalid file name')
```
Comment (thomasmunduchira) [inline src/controller/admin.py]: Should be 400

File: applications-frontend/src/components/Upload.vue (hunk 1 above)
Comment (klanmiko) [reply applications-frontend/src/components/Upload.vue]: name isn't used

File: src/config/config.py (hunk 4 above)
Comment (klanmiko) [reply src/config/config.py]: aiming for src/applicants

**Pull request merged**